pages/learn/develop/container-contracts.md

```diff
@@ -0,0 +1,112 @@
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-l4t-version'
+name: 'Enforce L4T Version'
+requires:
+ - type: 'sw.l4t'
+ version: '28.2'
+```
+
+* `type` should be set to `sw.container` for container contracts.
+* `slug` should be a unique identifier to identify the contract.
+* `name` should be a descriptive name of the contract.
+* `requires` should be a list of requirements to be enforced by the contract defined by `type` and `version` keys. Currently, `type` may be either `sw.supervisor` for the device Supervisor version or `sw.l4t` for the L4T version.
+
+__Note__ If `name`, `type`, or `slug` are omitted, the build with fail.
+
+Multiple requirements may be added to a single contract. The following example requires that the Supervisor version is greater than or equal to 10.6.13, and the L4T version is equal to 32.2.
+
+```yaml
+type: 'sw.container'
+slug: 'enforce-supervisor-and-l4t'
+name: 'Enforce supervisor and l4t requirements'
+requires:
+ - type: 'sw.supervisor'
+ version: '>=10.6.13'
+ - type: 'sw.l4t'
+ version: '32.2' 
+```
+
+__Note__ Any type other than `sw.supervisor` or `sw.l4t` will result in the contract being invalid.
+
+When deploying a release, if the contract requirements are not met, the release will fail, and all devices in the application will remain on their current release. This default behavior can be overridden by including an [optional label](#optional-containers) in the application docker-compose.yml file. This label specifies which services with unmet requirements will be ignored, with the release being deployed to all other devices in the application. If there are any existing running containers for services with unmet requirements, they will not be destroyed and continue running the previous release.
+
+Should a contract fail, the dashboard logs will contain detail about the failing contract:
+
+```shell
+Could not move to new release: Some releases were rejected due to having unmet requirements:
+Contracts: Services with unmet requirements: first-service
+```
+
+__Note__ For application fleets that contain a mixture of Supervisor and L4T versions, contracts may be used to ensure only compatible devices run specific services as releases will only be deployed to those meeting the contract requirements.
+
+## Optional containers
+
+By default, when a service contract fails, none of the services are deployed to the device. However, in a multi-container application, it is possible to ignore those services that fail the contract requirements with the other services being deployed as normal. To do so, we make use of the `io.balena.features.optional=1` [Supervisor label][labels] to indicate which services should be considered optional.
```

Comment:
```suggestion
By default, when a service contract fails, none of the services are deployed to the device. However, in a multi-container application, it is possible to ignore those services that fail the contract requirements with the other services being deployed as normal. To do so, we make use of the `io.balena.features.optional: 1` [Supervisor label][labels] to indicate which services should be considered optional.
```

Reply:
I think both styles work but will update for consistency with labels in the docs 👍